crates/engine/src/response/object_set.rs

```diff
@@ -69,7 +69,7 @@ pub(crate) struct InputResponseObjectSet {
 impl InputResponseObjectSet {
     pub(crate) fn with_response_objects(mut self, refs: Arc<ResponseObjectSet>) -> Self {
         let n = self.indices.len();
-        self.indices.reserve(refs.len());
+        self.indices.reserve_exact(refs.len());
```

Comment:
Just curious, why reserve_exact here?

Reply:
well, premature optimization to be honest. will remove it. We'll never write more in this function than `refs.len()` and it's unlikely to call this function multiple times. But who knows 🤷

Reply:
I was wondering because maybe it's pointing a flaw in my understanding. I thought there wasn't a performance advantage to reserve_exact(), and it was just a way to fine-tune the capacity without really an upside. Looking at the docs, I didn't know that part

> unlike [reserve](https://doc.rust-lang.org/std/vec/struct.Vec.html#method.reserve), this will not deliberately over-allocate to speculatively avoid frequent allocations.

https://doc.rust-lang.org/std/vec/struct.Vec.html#method.reserve_exact

So my mental model was wrong. I'd use reserve_exact() here too given that.